### PR TITLE
Implementing Pagination and Infinite Scrolling in Next.js

### PR DESCRIPTION
**Pull request by Tammibriggs**
Hi @phazonoverload, I am done with the first draft for issue #194

### CODE REVIEW

**Comment by phazonoverload** [comment]
Thanks for your patience. I just wanted to let you know that I'll be reviewing this in the next couple of days.

**Comment by Tammibriggs** [comment]
Awesome!

File: implementing-pagination-and-infinite-scrolling-in-nextjs/index.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+
+In Directus the second parameter can be either [offset](https://docs.directus.io/reference/query.html#offset) and [page](https://docs.directus.io/reference/query.html#page) which are associated with offset-based and page-based pagination respectively.
+
+- Offset-based pagination: This deals with specifying a value (or offset) that indicates the number of items to skip or where the items being fetched should start from. It works hand-in-hand with the `limit` parameter to return a fixed number of items.
+  For example, for a dataset of 200 items, if `offset=20` and `limit=10`, items 21-30 will be returned.
+
+- Page-base pagination: This is an abstraction of the offset-based pagination where a page number is specified (e.g. 1, 2, 3,…) which will be used under the hood to calculate the offset.
+
+For this tutorial, the `page` parameter will be used to implement both the pagination and infinite scrolling.
```
Comment (phazonoverload) [inline implementing-pagination-and-infinite-scrolling-in-nextjs/index.md:93]:
```suggestion
## Fetching Data in Chunks

To fetch data in chunks using an API, usually, two parameters are required. The first is a `limit` that determines the maximum number of items returned, and the second is a parameter that determines the starting point of the items to be fetched, it is usually associated with a type of pagination (e.g. cursor-based pagination, offset-based, pagination, etc.).

In Directus the second parameter can be either [offset](https://docs.directus.io/reference/query.html#offset) and [page](https://docs.directus.io/reference/query.html#page) which are associated with offset-based and page-based pagination respectively.

- Offset-based pagination: This deals with specifying a value (or offset) that indicates the number of items to skip or where the items being fetched should start from. It works hand-in-hand with the `limit` parameter to return a fixed number of items. For example, for a dataset of 200 items, if `offset=20` and `limit=10`, items 21-30 will be returned.
- Page-based pagination: This is an abstraction of the offset-based pagination where a page number is specified (e.g. 1, 2, 3) which will be used under the hood to calculate the offset.

For this tutorial, the `page` parameter will be used to implement both the pagination and infinite scrolling.
```

File: implementing-pagination-and-infinite-scrolling-in-nextjs/index.md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+| Site footer                      | With pagination the footer can be placed right were it ought to be — at the bottom of the page.                                                                                                                                                                            | With infinite scrolling not appropriate to place a footer at the bottom of the page since it can be overwhelming for users to scroll to the very bottom of the page. <br>Some alternatives to this included placing the footer links in the sidebar or adding a load more button for users to click to fetch more content rather than fetching at scroll. |
```
Comment (phazonoverload) [inline implementing-pagination-and-infinite-scrolling-in-nextjs/index.md:501]: Please can you make this prose instead of a table?

**Review by phazonoverload — CHANGES_REQUESTED** [review]
Really good post - I enjoyed reading about the approaches. i've left some comments and suggestions inline, but I have some additional post-wide changes I'd like you to make:

1. Make sure the Markdown formatting is correct, especially around paragraphs. You need to have an empty line between two paragraphs. 
2. Make sure you replace Nextjs with Next.js.
3. Whenever referring to a file, directory, or code (properties, field names, etc) surround them in backticks, not italics.
4. Please remove all of the Tailwind classes. It's distracting from the core point you're making. You can use styled screenshots, vut don't include them in instructions.
5. For the pagination, can you just show the active class being applied - no need to style it.

File: implementing-pagination-and-infinite-scrolling-in-nextjs/index.md (hunk 2 above)
Comment (Tammibriggs) [reply implementing-pagination-and-infinite-scrolling-in-nextjs/index.md:501]: Okay. I will do that

**Comment by Tammibriggs** [comment]
Hi @phazonoverload, thanks for the compliment. 

I have made the requested changes.

**Comment by Tammibriggs** [comment]
@phazonoverload I also made some modifications to the "When to choose between Pagination and Infinite scrolling" section in addition to the once requested.

**Comment by phazonoverload** [comment]
> I would like to point out that for the infinite scrolling due to the removal of the styling, the contents no longer span the entire height of the viewport, causing the next content to be automatically fetched without scrolling.

If it's functional leave it in

**Comment by Tammibriggs** [comment]
@phazonoverload Done.

**Comment by phazonoverload** [comment]
Thank you for writing this, and we are happy to consider it done in it's current state. My team will schedule it in to be published. 

This repo is just for authoring/reviewing - the actual piece will be published via a Directus project, so we'll take it from here. I'll close this issue to keep the main branch clean, but this is a acceptance of your work.

Could you email me at [devrel@directus.io](mailto:devrel@directus.io) and we'll share what we need to get you paid. Thanks!

**Pull request closed without merging**